api/integreatly/v1alpha1/grafanadatasource_types.go

```diff
@@ -168,6 +168,13 @@ type GrafanaDataSourceJsonData struct {
 	GithubUrl string `json:"githubUrl,omitempty"`
 	// Fields for Alertmanager data sources
 	Implementation string `json:"implementation,omitempty"`
+	// Fields for AWS Prometheus data sources
+	SigV4Auth          bool   `json:"sigV4Auth,omitempty"`
+	SigV4AuthType      string `json:"sigV4AuthType,omitempty"`
+	SigV4ExternalId    string `json:"sigV4ExternalId,omitempty"`
+	SigV4AssumeRoleArn string `json:"sigV4AssumeRoleArn,omitempty"`
+	SigV4Region        string `json:"sigV4Region,omitempty"`
+	SigV4Profile       string `json:"sigV4Profile,omitempty"`
```

Comment:
Remove these changes and create a separate PR.
Then we can fix the bug in this PR only.

Reply:
I see these as part of the bug.  Enabling SigV4Auth without being able to configure it in GrafanaDataSource is effectively a noop for folks that need to use it.